### ISSUE TITLE
Trial: possibly incorrect  error: lambda lists are not congruent.

### ISSUE THREAD

**Issue by kpoeck** [clasp-developers/clasp#733]
consider
```lisp
(in-package :cl-user)

;;; from https://github.com/Shirakumo/trial.git assets/image.lisp

(defgeneric load-image (path type &key width height pixel-type pixel-format &allow-other-keys))

;;; works fine
(defmethod load-image (path (type (eql :r16)) &rest args)
  (list path type args))

;;; errors
(defmethod load-image (path (type (eql :tga)) &key)
  (list path type))

Condition of type: SIMPLE-ERROR
Cannot add the method #<STANDARD-METHOD UNNAMED (T (EQL :TGA))> to the generic function #<STANDARD-GENERIC-FUNCTION LOAD-IMAGE> because their lambda lists (PATH TYPE &KEY) and (PATH TYPE &KEY WIDTH HEIGHT PIXEL-TYPE PIXEL-FORMAT &ALLOW-OTHER-KEYS) are not congruent.

Available restarts:
(use :r1 to invoke restart 1)
````
Works fine with
```lisp
(defmethod load-image (path (type (eql :tga)) &key #+clasp &allow-other-keys)
  (list path type))
````
As just discussed the relevant part of the spec is here:
http://www.lispworks.com/documentation/HyperSpec/Body/07_fd.htm
7.6.4 Congruent Lambda-lists for all Methods of a Generic Function
...
5. The use of &allow-other-keys need not be consistent across lambda lists. If &allow-other-keys is mentioned in the lambda list
of any applicable method or of the generic function, any keyword arguments may be mentioned in the call to the generic function.

One might understand 5. that &allow-other-keys  should not be checked.

**Comment by kpoeck** [comment]
Other lisps:
ecl: same error as clasp
ccl:
`;Compiler warnings for "home:lisp;compiler;clasp;congruent-methods.lisp.newest" :
;   In an anonymous lambda form at position 284: (:METHOD LOAD-IMAGE (T (EQL :TGA))) does not accept keywords (:WIDTH :HEIGHT :PIXEL-TYPE :PIXEL-FORMAT) required by the generic functions
` 
sbcl: no error, no warning
cmucl: no error, no warning
clisp
`
#<STANDARD-METHOD (#1=#<BUILT-IN-CLASS T> (EQL :TGA))> does not accept the keywords (:WIDTH
 :HEIGHT :PIXEL-TYPE :PIXEL-FORMAT) of #2=#<STANDARD-GENERIC-FUNCTION
     LOAD-IMAGE>
   [Condition of type SIMPLE-ERROR]
`
lispworks: no error, no warning

**Comment by kpoeck** [comment]
ECL fixed that in https://gitlab.com/embeddable-common-lisp/ecl/-/commit/0e6016559ecc914f1f83c1fb2bd19ab0cbee1508

**Comment by kpoeck** [comment]
Testing this as a fix in clasp

**Comment by kpoeck** [comment]
Fixed by using ecl fix from above in #1058
